DependencyInjection/ScopedContainer.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JMS\SerializerBundle\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Alias;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+final class ScopedContainer
+{
+    private $container;
+    private $instance;
+
+    public function __construct(ContainerBuilder $container, string $instance)
+    {
+        $this->container = $container;
+        $this->instance = $instance;
+        $container->setAlias('jms_serializer.instances.' . $instance, new Alias($this->getDefinitionRealId('jms_serializer.serializer'), true));
+    }
+
+    public function getInstanceName(): string
+    {
+        return $this->instance;
+    }
+
+    public function removeDefinition($id)
+    {
+        $this->container->removeDefinition($this->getDefinitionRealId($id));
+    }
+
+    public function register(string $id, ?string $class = null)
+    {
+        return $this->setDefinition($id, new Definition($class));
+    }
+
+    public function findDefinition(string $id): Definition
+    {
+        return $this->container->findDefinition($this->getDefinitionRealId($id));
+    }
+
+    public function findTaggedServiceIds($tag): array
+    {
+        $serviceIds = [];
+        foreach ($this->container->findTaggedServiceIds($tag) as $id => $tags) {
+            $def = $this->container->findDefinition($id);
+
+            if ($def->hasTag('jms_serializer.instance')) {
+                if ($def->getTag('jms_serializer.instance')[0]['name'] !== $this->instance) {
+                    continue;
+                }
+            }
+
+            foreach ($tags as $attributes) {
+                if (empty($attributes['instance']) || $attributes['instance'] === $this->instance) {
+                    $serviceIds[$id][] = $attributes;
+                }
+            }
+        }
+
+        return $serviceIds;
+    }
+
+    private function getDefinitionRealId($id): string
```

Comment:
`getRealId` requires string as second parameter, so it should be also required here I guess.
```suggestion
    private function getDefinitionRealId(string $id): string
```